crates/audio/src/whistle_detection.rs

```diff
@@ -20,8 +21,11 @@ pub const NUMBER_OF_AUDIO_CHANNELS: usize = 4;
 pub const NUMBER_OF_AUDIO_SAMPLES: usize = 2048;
 const NUMBER_OF_FREQUENCY_SAMPLES: usize = NUMBER_OF_AUDIO_SAMPLES / 2;
 
+#[derive(Deserialize, Serialize)]
 pub struct WhistleDetection {
+    #[serde(skip, default = "deserialize_not_implemented")]
```

Comment:
what do you think of https://doc.rust-lang.org/std/macro.unimplemented.html instead of `deserialize_not_implemented`?

Reply:
```
error[E0423]: expected function, found macro `unimplemented`
  --> crates/audio/src/whistle_detection.rs:26:29
   |
26 |     #[serde(skip, default = "unimplemented")]
   |                             ^^^^^^^^^^^^^^^ not a function
   |
help: use `!` to invoke the macro
   |
26 |     #[serde(skip, default = "unimplemented"!)]
   |                                            +
```

Reply:
Same with `panic!()`. I thought of something like [`panic_any()`](https://doc.rust-lang.org/std/panic/fn.panic_any.html) function but I need a function `fn() -> T`.

Reply:
Then at least use it in your custom function?

Reply:
Using it now. I thought you wanted to use `unreachable!()` which I would argue against. But `unimplemented!()` is fine for me.